elytron/src/main/java/org/wildfly/extension/elytron/ElytronDefinition.java

```diff
@@ -363,7 +363,7 @@ protected void revertUpdateToRuntime(OperationContext context, ModelNode operati
 
     @Override
     public void registerAdditionalRuntimePackages(ManagementResourceRegistration resourceRegistration) {
-        resourceRegistration.registerAdditionalRuntimePackages(RuntimePackageDependency.required("org.wildfly.security.elytron"));
+        resourceRegistration.registerAdditionalRuntimePackages(RuntimePackageDependency.required("org.wildfly.security.elytron"), RuntimePackageDependency.required("org.wildfly.security.elytron-dynamic-ssl"));
```

Comment:
I missed that one, `RuntimePackageDependency.required` is not capability aware, so it is going to require the "org.wildfly.security.elytron-dynamic-ssl" when we are running in a higher level capability than community, e.g default.

Should register this module conditionally based on the current capability level where this module is needed. 
We could evolve the RuntimePackageDependency.required() to accept a Stability level and implement the dynamic check on that method, but that is out of the scope of this PR

Reply:
@yersan I fixed this with:

```
if (resourceRegistration.getStability().enables(Stability.COMMUNITY)) {
            resourceRegistration.registerAdditionalRuntimePackages(RuntimePackageDependency.required("org.wildfly.security.elytron-dynamic-ssl"));
 }
```

Thank you!

Reply:
More on this, since it has been discussed also in Zulip: https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/Disable.20Specific.20Stability.20Levels.20at.20Provisioning.20Time

As originally mentioned at https://github.com/wildfly/wildfly-core/pull/5947#discussion_r1560868361, the requirement for this module should be done when the  Dynamic SSL resource is being added. That will remove the Stability level conditional check, since the Dynamic SSL resource will be only added when the server is launched at `community` level or lower levels.

Reply:
I think there is still more discussion to go - I think we are nearing the point the module is unlikely to be provisioned at all unless the provisioned configuration contains a definition of the resource but that is not what we are trying to achieve.

Reply:
In any case, @Skyllarr notice that what you are checking there is if the resourceRegistration enables the `community` stability. That's not what we would need to check, we would need to check whether the resourceRegistration enables the current stability where the server is running at runtime.

So, if you launch your server with `default` stability, your additional package won't be required to be registered.

Reply:
In any case, as discussed in Zulip, we should avoid this check and move the additional package registration to the child resource that requires it.

Reply:
Thanks @yersan , I've moved the module registration  to the `registerAdditionalRuntimePackages` method when registering dynamic ssl context